presto-common/src/main/java/com/facebook/presto/common/predicate/Range.java

```diff
@@ -54,6 +56,26 @@ public Range(
         this.high = high;
     }
 
+    // Private constructor to allow creating ranges without comparing markers.
+    // Only used to create canonicalized ranges with REMOVE_CONSTANTS strategy.
+    private Range(Type type, Marker low, Marker high)
+    {
+        requireNonNull(type, "value is null");
+        requireNonNull(low, "value is null");
+        requireNonNull(high, "value is null");
+        if (!low.getType().equals(high.getType())) {
+            throw new IllegalArgumentException(String.format("Marker types do not match: %s vs %s", low.getType(), high.getType()));
+        }
+        if (low.getBound() == Marker.Bound.BELOW) {
+            throw new IllegalArgumentException("low bound must be EXACTLY or ABOVE");
+        }
+        if (high.getBound() == Marker.Bound.ABOVE) {
+            throw new IllegalArgumentException("high bound must be EXACTLY or BELOW");
+        }
+        this.low = low;
+        this.high = high;
+    }
+
```

Comment:
Nah.... there is a better way....

The rule of thumb: **Always only have one main constructor. All other constructors should use `this` to avoid duplicated code. Duplicated code can cause inconsistent behavior.**


```java
@JsonCreator
public Range(
        @JsonProperty("low") Marker low,
        @JsonProperty("high") Marker high)
{
    this(low, high, () -> {
        if (low.compareTo(high) > 0) {
            throw new IllegalArgumentException("low must be less than or equal to high");
        }
    });
}

private Range(Marker low, Marker high, Runnable extraCheck)
{
    requireNonNull(low, "value is null");
    requireNonNull(high, "value is null");
    if (!low.getType().equals(high.getType())) {
        throw new IllegalArgumentException(String.format("Marker types do not match: %s vs %s", low.getType(), high.getType()));
    }
    if (low.getBound() == Marker.Bound.BELOW) {
        throw new IllegalArgumentException("low bound must be EXACTLY or ABOVE");
    }
    if (high.getBound() == Marker.Bound.ABOVE) {
        throw new IllegalArgumentException("high bound must be EXACTLY or BELOW");
    }
    extraCheck.run();
    this.low = low;
    this.high = high;
}